model/manifest_test.go

```diff
@@ -17,6 +17,163 @@ import (
+		}}, true},
+		{"Happy case", &Manifest{
+			Id:               "com.company.test",
+			Name:             "thename",
+			Description:      "thedescription",
+			HomepageURL:      "http://someurl.com",
+			SupportURL:       "http://someotherurl.com",
+			Version:          "0.0.1",
+			MinServerVersion: "5.6.0",
+			Server: &ManifestServer{
+				Executable: "theexecutable",
+			},
+			Webapp: &ManifestWebapp{
+				BundlePath: "thebundlepath",
+				BundleHash: []byte{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15},
+			},
+			SettingsSchema: &PluginSettingsSchema{
+				Header: "theheadertext",
+				Footer: "thefootertext",
+				Settings: []*PluginSetting{
+					{
+						Key:         "thesetting",
+						DisplayName: "thedisplayname",
+						Type:        "dropdown",
+						HelpText:    "thehelptext",
+						Options: []*PluginOption{
+							{
+								DisplayName: "theoptiondisplayname",
+								Value:       "thevalue",
+							},
+						},
+						Default: "thedefault",
+					},
+				},
+			},
+		}, false},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.Title, func(t *testing.T) {
+			err := tc.manifest.IsValid()
+			if tc.ExpectError {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestIsValidSettingsSchema(t *testing.T) {
+	testCases := []struct {
+		Title          string
+		settingsSchema *PluginSettingsSchema
+		ExpectError    bool
+	}{
+		{"Invalid Setting", &PluginSettingsSchema{Settings: []*PluginSetting{{Type: "invalid"}}}, true},
+		{"Happy case", &PluginSettingsSchema{Settings: []*PluginSetting{{Type: "text"}}}, false},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.Title, func(t *testing.T) {
+			err := tc.settingsSchema.isValidSchema()
+			if tc.ExpectError {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestSettingIsValid(t *testing.T) {
+	testCases := []struct {
+		Title       string
+		setting     *PluginSetting
+		ExpectError bool
+	}{
+		{"Nil setting", nil, false},
+		{"Invalid setting type", &PluginSetting{Type: "invalid"}, true},
+		{"RegenerateHelpText error", &PluginSetting{Type: "text", RegenerateHelpText: "some text"}, true},
+		{"Placeholder error", &PluginSetting{Type: "bool", Placeholder: "some text"}, true},
+		{"Nil Options", &PluginSetting{Type: "bool"}, false},
+		{"Options error", &PluginSetting{Type: "generated", Options: []*PluginOption{}}, true},
+		{"Options displayName error", &PluginSetting{Type: "radio", Options: []*PluginOption{
+			{
+				Value: "some value",
+			},
+		}}, true},
+		{"Options value error", &PluginSetting{Type: "radio", Options: []*PluginOption{
+			{
+				DisplayName: "some name",
+			},
+		}}, true},
+		{"Happy case", &PluginSetting{Type: "radio", Options: []*PluginOption{
+			{
+				DisplayName: "Name",
+				Value:       "value",
+			},
+		}}, false},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.Title, func(t *testing.T) {
+			err := tc.setting.isValid()
```

Comment:
This is causing a panic in the `nil` case cause you are calling a method on the `nil` pointer.

Reply:
My apologies. Did not run tests locally after making the last change.